src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
             }
+        }
+    }
 
-            std::thread *workerThreads = new std::thread[2];
-            workerThreads[0] =
-                std::thread(&JasmineGraphInstanceService::createPartitionFiles, graphID, partitionID, "local");
-            workerThreads[1] =
-                std::thread(&JasmineGraphInstanceService::createPartitionFiles, graphID, partitionID, "centralstore");
-
-            workerThreads[0].join();
-            instance_logger.log("WorkerThread 0 joined", "info");
-
-            workerThreads[1].join();
-            instance_logger.log("WorkerThread 1 joined", "info");
+    return egonetMap;
+}
 
-            write(connFd, JasmineGraphInstanceProtocol::SEND_PARTITION_ITERATION.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_PARTITION_ITERATION.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_PARTITION_ITERATION, "info");
+void calculateEgoNet(string graphID, string partitionID, int serverPort, JasmineGraphHashMapLocalStore localDB,
+                     JasmineGraphHashMapCentralStore centralDB, string workerList) {
+    std::vector<string> workerSockets;
+    stringstream wl(workerList);
+    string intermediate;
+    while (getline(wl, intermediate, ',')) {
+        workerSockets.push_back(intermediate);
+    }
+    map<long, map<long, unordered_set<long>>> egonetMap =
+        calculateLocalEgoNet(graphID, partitionID, serverPort, localDB, centralDB, workerSockets);
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partIteration(data);
+    string instanceDataFolderLocation = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+    string attributeFilePart = instanceDataFolderLocation + "/" + graphID + "_egonet_" + partitionID;
+    ofstream partfile;
+    partfile.open(attributeFilePart, std::fstream::trunc);
+    for (map<long, map<long, unordered_set<long>>>::iterator it = egonetMap.begin(); it != egonetMap.end(); ++it) {
+        map<long, unordered_set<long>> egonetInternalMap = it->second;
+        for (map<long, unordered_set<long>>::iterator itm = egonetInternalMap.begin(); itm != egonetInternalMap.end();
+             ++itm) {
+            unordered_set<long> egonetInternalMapEdges = itm->second;
+            for (unordered_set<long>::iterator ite = egonetInternalMapEdges.begin();
+                 ite != egonetInternalMapEdges.end(); ++ite) {
+                partfile << to_string(it->first) << "\t" << to_string(itm->first) << "\t" << to_string(*ite) << endl;
+            }
+        }
+    }
+    partfile.close();
 
-            write(connFd, JasmineGraphInstanceProtocol::SEND_PARTITION_COUNT.c_str(),
-                  JasmineGraphInstanceProtocol::SEND_PARTITION_ITERATION.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_PARTITION_COUNT, "info");
+    // todo  invoke other workers asynchronously
```

Comment:
I hope this todo statement has not been added by you @thevindu-w

Reply:
It was there in the original code also

Reply:
Ok